pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1547,6 +1587,86 @@ func useCSIPlugin(vpm *volume.VolumePluginMgr, spec *volume.Spec) bool {
 	return false
 }
 
+func nodeUsingCSIPlugin(og *operationGenerator, spec *volume.Spec, nodeName types.NodeName) (bool, error) {
+	var err error
+
+	migratable, err := og.volumePluginMgr.IsPluginMigratableBySpec(spec)
+	if err != nil {
+		return false, err
+	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIMigration) ||
+		!utilfeature.DefaultFeatureGate.Enabled(features.CSINodeInfo) ||
+		!migratable {
+		return false, nil
+	}
+
+	if len(nodeName) == 0 {
+		return false, fmt.Errorf("nodeName is empty")
+	}
+
+	vpm := og.volumePluginMgr
+
+	kubeClient := vpm.Host.GetKubeClient()
+	if kubeClient == nil {
+		// TODO(dyzz) check this error case, what should we do in standalone kubelet mode?
+		return false, fmt.Errorf("failed to get kube client from volume host")
```

Comment:
`errors.New`

Reply:
Is there a reason for using `errors.New` vs. `fmt.Errorf` in this case besides personal preference?
for some of these files we would have to import a "different" error package since the default is the apimachinery one which does not contain `errors.New`. This also makes it slightly more difficult if we want to augment errors in the future to have more information.

Stylistically I prefer `fmt.Errorf` but curious to hear if you have a more compelling reason to switch

Reply:
overall this is a minor nit.

but errors.New is many times faster than fmt.Errorf , because Errorf uses Sprintf.
https://golang.org/src/fmt/print.go?s=6105:6155#L212
https://golang.org/src/errors/errors.go?s=293:320#L1

if errors are non fatal this can stall the CPU.

Reply:
no worries, makes sense to me I will make the changes